### PR TITLE
Move deprecated DDSs from dds/sequence to new packages

### PR DESCRIPTION
**Pull request by alex-pardes**
## Description

This change moves deprecated DDSs, namely `SharedNumberSequence`, `SharedObjectSequence`, and `SparseMatrix`, out of `@fluidframework/sequence` to a new package, `@fluid-experimental/sequence-deprecated`.

## Breaking changes

This change will require users of the aforementioned DDSs to import them from the new package.

### CODE REVIEW

File: experimental/dds/sequence-deprecated/package.json [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    "@fluidframework/protocol-definitions": "^0.1029.1000-0",
+    "@fluidframework/runtime-definitions": "^2.0.0",
+    "@fluidframework/runtime-utils": "^2.0.0",
+    "@fluidframework/sequence": "^2.0.0",
+    "@fluidframework/shared-object-base": "^2.0.0",
+    "@fluidframework/telemetry-utils": "^2.0.0",
+    "uuid": "^8.3.1"
+  },
+  "devDependencies": {
+    "@fluid-internal/stochastic-test-utils": "^2.0.0",
```
Comment (Abe27342) [inline experimental/dds/sequence-deprecated/package.json:81]: probably worth doing a pass over devDeps and deps--I don't think this package will be referenced by any moved code.

File: experimental/dds/sequence-deprecated/src/test/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+    "extends": "@fluidframework/build-common/ts-common-config.json",
+    "compilerOptions": {
```
Comment (Abe27342) [inline experimental/dds/sequence-deprecated/src/test/tsconfig.json:3]: FYI: #10957 turns on strict null checks in sequence. Dunno which will merge first, but eventually we should have it on here as well.

Comment (Abe27342) [reply experimental/dds/sequence-deprecated/src/test/tsconfig.json:3]: well, at least in the non-test config :)

File: packages/dds/default-map/.vscode/launch.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "name": "Sequence Test",
+            "type": "node",
+            "request": "launch",
+            "program": "${workspaceRoot}/../../../node_modules/mocha/bin/_mocha",
+            "args": ["dist/test", "--no-timeouts", "--exit"],
+            "cwd": "${workspaceRoot}",        }
```
Comment (Abe27342) [inline packages/dds/default-map/.vscode/launch.json:13]: nit: weird formatting of trailing }

File: packages/dds/default-map/package.json [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+  "types": "dist/index.d.ts",
+  "scripts": {
+    "build": "npm run build:genver && concurrently npm:build:compile npm:lint && npm run build:docs",
+    "build:commonjs": "npm run tsc && npm run typetests:gen && npm run build:test",
+    "build:compile": "concurrently npm:build:commonjs npm:build:esnext",
+    "build:docs": "api-extractor run --local --typescript-compiler-folder ../../../node_modules/typescript && copyfiles -u 1 ./_api-extractor-temp/doc-models/* ../../../_api-extractor-temp/",
+    "build:esnext": "tsc --project ./tsconfig.esnext.json",
+    "build:full": "npm run build",
+    "build:full:compile": "npm run build:compile",
+    "build:genver": "gen-version",
```
Comment (Abe27342) [inline packages/dds/default-map/package.json:25]: genver and associated scripts shouldn't be necessary for this package: I wouldn't expect it to have any code that requires meta-knowledge about the version. (that's true for actual DDSes due to the "attributes" static prop on their factory)

File: packages/dds/default-map/src/packageVersion.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ *
+ * THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY
+ */
+
+export const pkgName = "@fluidframework/default-map";
```
Comment (Abe27342) [inline packages/dds/default-map/src/packageVersion.ts:8]: after deleting `genver` script, also remove this file

File: api-report/default-map.api.md [hunk 6]
```diff
@@ -0,0 +1,140 @@
+## API Report File for "@fluidframework/default-map"
```
Comment (Abe27342) [inline api-report/default-map.api.md:1]: Putting it here for lack of a better place: PR should update `BREAKING.md`

**Review by Abe27342 — COMMENTED** [review]
Generally looks good--mostly superficial feedback. Need some basic READMEs as well.

File: packages/dds/default-map/package.json [hunk 7]
```diff
@@ -0,0 +1,106 @@
+{
+  "name": "@fluidframework/default-map",
+  "version": "2.0.0",
+  "description": "Distributed map",
```
[the remark this answers is not in the document]
Comment (anthony-murphy) [reply packages/dds/default-map/package.json:4]: i'm pretty against adding a new package here? do we really need it?

File: experimental/dds/sequence-deprecated/src/sharedIntervalCollection.ts [hunk 8]
```diff
@@ -76,11 +78,6 @@ export class SharedIntervalCollectionFactory implements IChannelFactory {
     }
 }
 
-export interface ISharedIntervalCollection<TInterval extends ISerializableInterval> {
```
Comment (anthony-murphy) [inline experimental/dds/sequence-deprecated/src/sharedIntervalCollection.ts:79]: let's not move this in this pr, as i think this is what is force us to also break default map into a new package

**Comment by anthony-murphy** [comment]
in general i don't think there is an advantage to doing everything at once here. i'd start with the simplest to move, and just use that to get the new package setup and in, then we can work through the over moves one by one

**Review by Abe27342 — APPROVED** [review]
:shipit:

**Pull request merged**